### DISCUSSION TITLE
Regarding quarkus 3.7.4 compatibility with Oracle Graalvm Enterprise Edition 21.3.8

### DISCUSSION THREAD

**Discussion by mehab** [quarkusio/quarkus#39324] [Q&A]
Hey Team,

I am currently using Quarkus version 3.7.4 for my cli application and trying to build native image using Oracle Graalvm Enterprise edition 21.3.8 (released in October 2023) and the native image build step is failing with this error: io.quarkus.deployment.pkg.steps.nativeimagebuildstep threw an exception java.lang.illegalStateException Out of date version of graalvm detected: 21.3.8
It is not possible to get onto enterprise edition of graalvm which has version 23.1.0 since it is not yet released according to the release calendar here: https://docs.oracle.com/en/graalvm/release-calendar.html Please can you help with this ? 
I am able to build with community edtion using mandrel image but this image is not allowed within my enterprise environment for official purposes so I need to use Enterprise edition Graalvm with LTS support.

[a comment by an automated account is not shown]

**Comment by zakkak** [comment]
Hi @mehab, please update to "GraalVM for JDK 21.0.2" whose internal version is `23.1.2` (which what Quarkus refers to) and let us know if it works.

**Reply by mehab** [reply]
![java and native image version](https://github.com/quarkusio/quarkus/assets/26340948/936c452f-4d49-41ea-bbd6-e107b8201862)
![native-image-stepStuck](https://github.com/quarkusio/quarkus/assets/26340948/61e53ffc-0198-4d84-9cc7-09d4c4d2cb22)
Hey @zakkak  I tried with the Graalvm for jdk 21.0.2. Attaching the screen shots of version and native-image version commands. But the command mvn package -Pnative -DskipTests command is getting stuck.

**Reply by zakkak** [reply]
How long did you let it run? How much free memory has the machine when you try to build? `native-image` can be slow (not that slow that you won't even see a couple of lines before getting stuck though).

**Reply by mehab** [reply]
@zakkak it was running overnight and did not progress. Also checked the processes running with top command and there was no process related to java running. Please could you provide some way I could debug more?

**Reply by galderz** [reply]
Do you also see the same hanging when running locally? Can you share a reproducer?

**Reply by mehab** [reply]
Hey @galderz  sorry I dont have a local Linux machine with enterprise edition of graalvm. But I did also try with this example: https://github.com/quarkusio/quarkus-quickstarts/archive/main.zip -- this codebase-- quarkus-quickstarts-main\getting-started-command-mode and added this parameter to application.properties files of quarkus project: `quarkus.native.additional-build-args=-H:-CheckToolchain`
I think it is a graalvm related issue. But I wanted yours and your team's opinion to be 100 percent sure here.

**Reply by zakkak** [reply]
@mehab is this only happening with Oracle GraalVM? Is there any chance you could try with the community edition of GraalVM for JDK 21.0.2 or Mandrel 23.1.2? If the issue only manifests when using Oracle GraalVM I suggest opening a GraalVM issue.
